Source/Core/Core/Boot/Boot_WiiWAD.cpp

```diff
@@ -2,37 +2,69 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
-#include <cstddef>
 #include <memory>
 #include <string>
 
-#include "Common/CommonPaths.h"
 #include "Common/CommonTypes.h"
-#include "Common/FileUtil.h"
 #include "Common/MsgHandler.h"
-#include "Common/NandPaths.h"
 
 #include "Core/Boot/Boot.h"
 #include "Core/CommonTitles.h"
 #include "Core/IOS/ES/ES.h"
 #include "Core/IOS/ES/Formats.h"
-#include "Core/IOS/FS/FileIO.h"
+#include "Core/IOS/FS/FileSystem.h"
 #include "Core/IOS/IOS.h"
 #include "Core/WiiUtils.h"
 
 #include "DiscIO/WiiWad.h"
 
+// cdb.vff is a virtual Fat filesystem created on first launch of sysmenu
+// we create it here as it is faster ~3 minutes for me when sysmenu does it
+// ~1 second created here
+static void CreateVirtualFATFilesystem(std::shared_ptr<IOS::HLE::FS::FileSystem> fs)
+{
+  constexpr u32 SYSMENU_UID = 0x1000;
+  constexpr u16 SYSMENU_GID = 0x1;
+
+  const std::string cdb_path = "/title/00000001/00000002/data/cdb.vff";
+  fs->CreateFile(SYSMENU_UID, SYSMENU_GID, cdb_path, 0, IOS::HLE::FS::Mode::ReadWrite,
+                 IOS::HLE::FS::Mode::ReadWrite, IOS::HLE::FS::Mode::ReadWrite);
+
+  constexpr size_t CDB_SIZE = 0x01400000;
+  const auto metadata = fs->GetMetadata(SYSMENU_UID, SYSMENU_GID, cdb_path);
+  if (!metadata || metadata->size >= CDB_SIZE)
+    return;
+
+  const auto fd = fs->OpenFile(SYSMENU_UID, SYSMENU_GID, cdb_path, IOS::HLE::FS::Mode::Write);
+  if (!fd)
+    return;
+
+  constexpr u8 CDB_HEADER[0x20] = {'V', 'F', 'F', 0x20, 0xfe, 0xff, 1, 0, 1, 0x40, 0, 0, 0, 0x20};
+  constexpr u8 CDB_FAT[4] = {0xf0, 0xff, 0xff, 0xff};
+  fs->WriteFile(*fd, std::vector<u8>(CDB_SIZE).data(), CDB_SIZE);
```

Comment:
Yes, this cannot be a sparse file anymore because the backend may not support them (the original filesystem -- SFFS does not). This shouldn't be a problem though, as it's just used to initialize the file to the correct size.

I'll look into replacing this series of writes and seeks with operations on a buffer as that is more efficient than writing zeroes.